### PR TITLE
feat(adapters): extract Backbone and jQuery integrations

### PR DESCRIPTION
**Pull request by paulfalgout**
## What

- Add the separately published `@marionette/adapters` package with explicit `./backbone` and `./dom/jquery` exports and no root barrel.
- Replace the mutating Backbone installer with one combined `BackboneApi` that applications pass to `setDataApi()` and `setStateApi()` on the selected runtime.
- Extend package, fixture, browser, performance, and release verification to require the adapter distribution as the third exact release artifact.

## Why

Marionette core should remain independent of optional Backbone and jQuery integrations. The prior core subpaths mixed optional runtime concerns into the main package and the Backbone installer modified Backbone prototypes, which could replace native listener behavior and made isolated runtime selection impossible.

## Agent-development failure addressed

- The first extraction draft retained the mutating installer. Source-freeze requirements and independent review caught that conflict before publication.
- The final adapter imports no Backbone or Marionette runtime, preserves listeners registered before configuration and caller-owned listeners, leaves Backbone namespaces and prototypes unchanged, and never calls persistence-capable `Backbone.Model#destroy()` during owned-state cleanup.
- Review also made copied ordered model snapshots explicit and added fail-closed coverage for the adapter npm action and exact Marionette peer version.

## Public behavior contract

- Canonical behavior after this PR: install `@marionette/adapters`, import the required explicit subpath, and pass the adapter to the selected runtime setter(s).
- Superseded behavior removed: `marionette/backbone` and `marionette/jquery-dom-api` no longer resolve; there are no aliases or forwarding modules.
- Compatibility exception: CI-only exact-base performance bridges are required for this relocation and are removed after it reaches the pull request base. No public runtime alias or fallback is retained.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [ ] Changes an existing production path
- [x] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: adapter aggregate is 1.46 kB Brotli: Backbone ESM 546 B, Backbone CJS 560 B, jQuery ESM 176 B, and jQuery CJS 174 B. Core remains free of Backbone and jQuery imports.
- Startup/hot path: no adapter code runs until a consumer imports and configures it. `BackboneApi.models()` returns a copied ordered snapshot rather than exposing Backbone's mutable internal array.
- Allocations/retention: the model snapshot allocates one shallow array copy per call. Adapter configuration adds no singleton listener store or retained model graph.

## Scope

- Affects optional Backbone data/state/event integration, optional jQuery DomApi integration, package exports and declarations, consumer fixtures, documentation, performance contracts, and release construction/promotion.
- Does not change native core defaults, publish any package, create a release, tag a commit, or deploy an application.

## Deployment Impact

- No deployment or package publication occurs in this PR.
- A future v5 release must publish the exact `marionette`, `@marionette/data`, and `@marionette/adapters` tarballs together through the verified release workflow.
- Existing pre-v5 consumers using the removed core adapter subpaths must update imports and explicit runtime configuration when adopting this head.

## Testing

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm test -- --reporter=dot                    # 110 files, 2,046 tests passed
npm run coverage                              # 100% statements/branches/functions/lines; 19 agent-contract tests passed
npm run test:source
npm run test:dist
npm run test:fixtures
npm run test:browser                          # adapter package and lifecycle cases passed in Chromium, Firefox, and WebKit
npm run test:performance                      # 120 tests passed
npm run size                                  # 89.86 kB cumulative / 100 kB ceiling; adapters 1.50 kB
npm run docs:check                            # 22 executable examples, 55 pages, 503 links
npm run lint:ci
npm run test:release-promotion                # 10 tests passed
npm run check:dist
npm run release:artifact -- --output <temp> --source-commit adc2442962f5a7878d373e1f8edaf844f6473f39 --source-ref refs/heads/feat/marionette-adapters --repository marionettejs/marionette
npm run release:verify -- --artifact-dir <temp> --source-commit adc2442962f5a7878d373e1f8edaf844f6473f39 --repository marionettejs/marionette
npm run release:github -- --mode dry-run --artifact-dir <temp>
npm run release:targets -- --mode dry-run --artifact-dir <temp>
npm publish <each exact tarball> --dry-run --ignore-scripts --access public --tag alpha
npm run test:fixtures -- --tarball <core> --data-tarball <data> --adapters-tarball <adapters>
```

The target dry run correctly reported the already-published core alpha and tag as conflicts while both scoped package versions are available. No publish was attempted.

## Package and browser impact

- Entry points or install fixtures affected: removes the two core adapter subpaths; adds `@marionette/adapters/backbone`, `@marionette/adapters/dom/jquery`, and `@marionette/adapters/package.json`.
- Browser contracts affected: packed adapter behavior is exercised in Chromium, Firefox, and WebKit, including native Backbone prototype and pre-configuration-listener preservation.
- Production/dev/test module-graph impact: core has no new runtime imports; the Backbone graph contains two internal modules and zero externals; the jQuery graph contains one internal module and one `jquery` external.

## Rollback or deprecation

- Revert this commit before publishing a v5 release. No compatibility aliases or runtime migrations need removal because none are introduced.

## Review notes

- Independent full-diff review found no remaining merge blockers after the snapshot and release-decision amendments.
- Historical performance evidence intentionally retains the old artifact names; current contracts and docs use the new package subpaths.
- The exact-base resource and consumer-fixture probes require one-PR, non-published compatibility bridges. They are pinned to this exact relocation, tested fail-closed, and do not alter the approval input.

---
## Summary by cubic
Extracts Backbone and jQuery integrations from core into a new separately published `@marionette/adapters` package. The old `marionette/backbone` side-effect import mutated Backbone prototypes and only configured the default runtime; the new `BackboneApi` is passed explicitly to `setDataApi()`/`setStateApi()`, leaves Backbone objects and prototypes untouched, and works with isolated runtimes.

**Migration**
- Install `@marionette/adapters` and import `BackboneApi` from `@marionette/adapters/backbone`, then call `setDataApi(BackboneApi)` and `setStateApi(BackboneApi)` at boot.
- Replace `marionette/jquery-dom-api` imports with `@marionette/adapters/dom/jquery` and pass the adapter to the selected runtime's `setDomApi()`.
- `marionette/backbone` and `marionette/jquery-dom-api` no longer resolve; no aliases are provided.
- The adapter returns copied ordered model snapshots, preserves listeners registered before configuration, rolls back a failed `subscribe` with `off()`, and owned-state disposal never calls persistence-capable `Model#destroy()`.
- Release verification now requires the adapters tarball as a third exact release artifact alongside `marionette` and `@marionette/data`.

<sup>Written for commit adc2442962f5a7878d373e1f8edaf844f6473f39. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/389?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added the optional `@marionette/adapters` package with Backbone and jQuery DOM integrations.
  * Backbone integration now supports explicit Data API and State API configuration without modifying Backbone objects or prototypes.
  * Added browser and TypeScript compatibility coverage for the adapters.

* **Changes**
  * Backbone and jQuery integrations moved to the new adapter package paths.
  * Legacy core adapter subpaths are no longer available.

* **Documentation**
  * Updated installation, migration, routing, events, and rendering guidance for the new setup.

### CODE REVIEW

**Comment by paulfalgout** [comment]
```json
{
  "schemaVersion": 1,
  "headSha": "adc2442962f5a7878d373e1f8edaf844f6473f39",
  "issueUrl": "https://github.com/marionettejs/marionette/issues/127",
  "approvedPaths": [],
  "approvedNewSubpaths": [
    "@marionette/adapters/backbone",
    "@marionette/adapters/dom/jquery"
  ],
  "approvedNewArtifacts": [
    {
      "path": "packages/adapters/dist/backbone.cjs",
      "size": 560
    },
    {
      "path": "packages/adapters/dist/backbone.js",
      "size": 546
    },
    {
      "path": "packages/adapters/dist/dom/jquery.cjs",
      "size": 174
    },
    {
      "path": "packages/adapters/dist/dom/jquery.js",
      "size": 176
    }
  ],
  "approvedTimingHarnessRevision": "20d861aca94353297814e6bd933b4d3379b842c5385fbe8bcd32eac6ce9f0c14",
  "evidenceUrls": [
    "https://github.com/marionettejs/marionette/issues/127#issuecomment-5524796135"
  ]
}
```

**Comment by paulfalgout** [comment]
Reviewed at fetched SHA `70d899c0` (base `bce8166f`). Core is now four artifacts totalling **88,406** Brotli-11 bytes, with the adapter artifacts relocated to `packages/adapters/dist/`.

**Approving the extraction**, with one correction of my own advice and one open question below.

### Correction: my `baselineModules` recommendation on #388 was wrong

I said the relocated-graph check should get "one more `isDeepStrictEqual` beside the one just added," matching `baselineExternalImports`. That would have been incorrect, and it would have blocked this PR. This relocation shows why:

```
./backbone          modules ['backbone.js']
@marionette/adapters/backbone   modules ['packages/adapters/src/backbone-api.js', 'packages/adapters/src/backbone.js']
```

`baselineExternalImports` are package specifiers — `backbone`, `marionette`, `underscore` — which legitimately survive a move unchanged, and they did here, exactly. `baselineModules` are repo-relative source paths, which *necessarily* change when the source relocates. Requiring equality would forbid the only thing a relocation is for. The asymmetry is principled, not an oversight, and I should have checked what the fields held before recommending they be treated alike.

### The residual concern is narrower, and this PR exercises it

`baselineModules` is still unconstrained for a relocated graph — not merely un-equal, but unchecked beyond `Array.isArray`. This relocation takes `./backbone` from **one** module to **two**, adding `backbone-api.js`.

That is plausibly correct here: combining the data and state adapters into one `BackboneApi` is the substance of the PR, and a new source file is the honest result. But the Phase-0 baseline for that graph grew during a move, and the gate cannot see it. For a *new* graph the contract demands empty baselines; for a moved one, any module list passes. A relocation is therefore the one route by which a production graph's module baseline can expand without review.

A proportionate check would be on **count** rather than contents — a relocated graph may rewrite its module paths but not grow its module set — leaving genuine additions to go through the new-graph path where they are visible. Worth considering before more packages relocate, since this is now the pattern.

### Open question: the consumer-fixture guard

On #381 I flagged that migrating the `v1` fixture in place would trip `base.fixtureRevision !== current.fixtureRevision`. That migration is exactly what happened — `contract.json`'s `sha256` moves from `cbb5cf51…` to `3bfbbe21…`, with scenario ids preserved and imports rewritten to `@marionette/adapters/backbone` and `@marionette/adapters/dom/jquery`, which was the right call given the base script hardcodes `v1`.

The guard at line 981 is unchanged, and consumer violations still reach `process.exitCode = 1`. I cannot see an exemption in this diff, so I would expect the consumer comparison to be refused on this PR specifically. If that is handled — a bootstrap allowance, a CI flag, something in the workflow I did not read — it is worth saying so in the PR body, because the next person to migrate a fixture will hit it and find no record.

### What this closes

The mutating installer is gone, and that resolves the finding I raised on #383: an isolated `createMarionette()` runtime could not receive the Backbone integration, because `import 'marionette/backbone'` configured the default runtime by prototype mutation. Passing `BackboneApi` to `setDataApi()`/`setStateApi()` on the selected runtime makes isolation work as documented, and removes the silent-wrong-answer path where a Backbone model in an isolated runtime read as an empty plain object.

Not calling persistence-capable `Backbone.Model#destroy()` during owned-state cleanup is the other quiet but important one — that is a network call in a teardown path.

### On the disclosure

Recording that the first draft retained the mutating installer, and that source-freeze review caught it before publication, is worth more than it costs. Given the roadmap now treats agent-assisted development as a measured capability rather than an implementation detail, a recorded near-miss is evidence about the workflow — and the useful direction of evidence, since it names what the review step actually caught.

Scope: I verified the relocation declaration, both graph baselines, the core artifact set and aggregate, and the fixture migration. I have **not** reviewed the adapter source, the release-verification changes, or the browser and packed fixtures.

**Comment by paulfalgout** [comment]
**Approval stands** at `b63adb62`.

### The base-authority loader is the right shape for a genuinely awkward problem

The exact-base resource probe imports `dist/backbone.js`, which this PR removes, and the base script is the one that executes — so there is no way for the candidate to teach it the new path. A scoped Node resolve hook that redirects that one specifier to a bridge configuring `BackboneApi` from the relocated build is a reasonable answer, and the comment is exactly what such a thing needs:

```
# Remove this non-published loader after the relocation reaches the PR base.
```

Self-limiting, with the removal condition stated rather than implied. This is the concrete form of the base-authority constraint I described on #377 — the base checkout's hardcoded paths execute against candidate content — and it will recur for every future relocation, so the pattern is worth having established.

### The name revert shows #388 working, and exposes a consequence

```diff
- "name": "Backbone adapter ES module",
+ "name": "Backbone shim ES module",
```

Master calls it `Backbone shim ES module`, so the relocated-artifact check — *"must preserve its exact-base name and baseline"* — forced this back. The guard did its job.

But look at what it froze. "Shim" described the **mutating installer that this PR removes**. The artifact is now a non-mutating adapter living at `packages/adapters/dist/backbone.js`, and it is required to carry a label that the same PR makes inaccurate.

It also cannot be fixed later. Once this lands, the name is in the exact-base contract, and any change trips *"removes or changes exact-base runtime artifact"*. The only escape is declaring another relocation, which requires `from !== to` — so correcting a display label would mean moving the file. These names are now permanently frozen, and #386's terminology normalization cannot reach them.

Worth considering whether `name` belongs in the frozen comparison at all. Artifact identity is `path` — every map in `growth-approval.mjs` is keyed by it, and the anti-laundering value comes from `baselineBrotliBytes` being preserved, not from the label. Excluding `name` from the equality checks would let it be corrected independently while leaving every guarantee intact. If it stays frozen, the alternative is to accept that these labels are permanent and get them right in this PR — but that is blocked by the very rule under discussion, which is the circularity worth naming.

Not a blocker. It is a small wart that becomes permanent at merge, which is the only reason I would rather raise it now than after.

**Comment by paulfalgout** [comment]
Addressed the consumer-fixture question at exact head `cdd9fe749ce070dc53e984c77e4bfbf8cd35408a`.

I reproduced the full exact-base report failure locally. The fix is a one-PR, fail-closed comparison bridge pinned to the exact old/new fixture digests, unchanged scenario/toolchain/peer metadata, and the exact adapter relocation map. The untouched authority report remains the growth-approval input; only the report comparison copy normalizes the fixture revision. The old authority resource loader also gets a non-published import bridge for its hardcoded `dist/backbone.js` path. Both are marked for removal after this relocation reaches the PR base. The focused regression plus the complete performance suite pass (120/120).

I am not adding a relocated-module count ceiling here. This legitimate extraction splits one old entry module into an entry plus `backbone-api.js`; the target graph and its two-module inventory are visible in the exact new-subpath approval. A count-only rule would reject that valid decomposition without distinguishing production cost or behavior, so that broader governance question should be evaluated separately rather than folded into this extraction.

**Comment by paulfalgout** [comment]
**Approval stands** at `cdd9fe74`. This addresses the consumer-fixture question I raised, and the shape of the answer is better than a general exemption would have been.

The bridge is pinned to one specific transition rather than to the *concept* of a fixture change:

```js
const expectedTransition = {
  from: 'cbb5cf51f81b78c74238111372eb9b7148b081b3b0d30a91baf0ef613724b134',
  to:   '3bfbbe21dbbec92e38439bc94df54c58c68c5c39ac554f961179e3b0f1c33f58',
};
```

Both fixture hashes and the full relocation map must match exactly, so this cannot be reused to wave through a later fixture migration — the next one will find the hashes disagree and be refused. That is the difference between a one-time bridge and a permanent hole, and it is the distinction I was worried about when I raised the guard on #381: an exemption written against "a fixture changed" would have quietly disarmed the check forever.

The CI wiring fails closed, which is the part worth confirming rather than assuming:

```bash
if node scripts/performance/bridge-consumer-fixture-transition.mjs …; then
  comparison_report="${PERFORMANCE_DIR}/bundle-size-comparison.json"
else
  bridge_status=$?
fi
```

On failure `comparison_report` stays pointed at the unbridged authority report **and** `bridge_status` feeds the `exit 1` condition, so a broken bridge cannot silently degrade into an unbridged comparison that happens to pass. `fails closed for any report or relocation drift` covers exactly that.

### One asymmetry with the other transition mechanism

The exact-base resource loader carries its removal condition in the CI file:

> `# Remove this non-published loader after the relocation reaches the PR base.`

The bridge has no equivalent note, in either the script or the workflow step. It is self-expiring in the sense that it refuses to apply to anything else — so it cannot do harm — but it will sit in `scripts/performance/` as permanently dead code once the relocation reaches the base, along with its test, and nothing records that it is finished. Both mechanisms exist for the same one-release window and it would be worth them being removable by the same signal.

A one-line comment matching the loader's would do it.

File: packages/adapters/src/backbone-api.js [hunk 1]
```diff
@@ -27,11 +32,18 @@ export default {
...
   },
 
   subscribe,
 
+  // Model#destroy may perform persistence, so owned state is released only
+  // through Backbone's native event cleanup methods.
+  disposeOwned(source) {
+    source.stopListening();
+    source.off();
+  },
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply packages/adapters/src/backbone-api.js:44]: Fixed at eb06770e. `disposeOwned` is now a no-op because Backbone has no source-wide disposal that preserves caller-owned listeners, while Marionette already invokes the adapter subscription disposer first. The regression now requires the external listener to fire after owner destruction; docs explicitly preserve caller ownership.

File: test/fixtures/backbone-adapter-types/validate.cts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import BackboneApi = require('@marionette/adapters/backbone');
+import Backbone = require('backbone');
+
+const model = new Backbone.Model({ id: 1, name: 'first' });
+const collection = new Backbone.Collection([model]);
+const listener = new Backbone.Model();
+
+listener.listenTo(model, 'change:name', (changedModel, value) => {
+  const typedModel: Backbone.Model = changedModel;
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/fixtures/backbone-adapter-types/validate.cts:9]: Verified without a cast. `npm run test:fixtures` compiles this exact fixture under strict TypeScript 7.0.2 with pinned `@types/backbone` 1.4.23 for both ESM and CJS. Since the callback argument already type-checks as written, adding a cast would weaken the fixture without fixing a reproducible failure.

File: test/fixtures/run.mjs [hunk 3]
```diff
@@ -107,6 +108,37 @@ try {
...
+    ]);
+    const packedAdaptersTarballs = readdirSync(packDir)
+      .filter(fileName => fileName.endsWith('.tgz') && !existingTarballs.has(fileName));
+    if (packedAdaptersTarballs.length !== 1) {
+      throw new Error(`Expected one packed adapters tarball, found ${packedAdaptersTarballs.length}`);
+    }
+    adaptersTarballPath = resolve(packDir, packedAdaptersTarballs[0]);
+  }
+
+  const adapterFixtures = new Set([
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/fixtures/run.mjs:133]: Added `adapters-package-vite` at eb06770e. It imports both adapter subpaths, configures an isolated runtime, uses Backbone data and the jQuery DomApi, and passes a Vite 5 production build from the exact packed core and adapters tarballs.

File: test/unit/backbone-adapter.spec.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+    owner.getState();
+    state.set('ready', true);
+    state.trigger('external');
+    owner.destroy();
+    state.set('ready', false);
+    state.trigger('external');
+
+    expect(onReady).to.have.been.calledOnce;
+    expect(externalListener).to.have.been.calledOnce;
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/backbone-adapter.spec.js]: Fixed at eb06770e. The post-destroy assertion now requires `externalListener` to have been called twice, so it proves owned cleanup preserves the caller-owned listener.

File: scripts/performance/load-runtime.mjs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+export async function loadBackboneRuntime(root, dependencyRoot = root) {
+  const requireFromRoot = createRequire(resolve(dependencyRoot, 'package.json'));
+  const Backbone = requireFromRoot('backbone');
+  const [Marionette, { default: BackboneApi }] = await Promise.all([
+    import(pathToFileURL(resolve(root, 'dist/marionette.js')).href),
+    import(pathToFileURL(resolve(root, 'packages/adapters/dist/backbone.js')).href),
+  ]);
+
+  Marionette.setDataApi(BackboneApi);
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply scripts/performance/load-runtime.mjs:13]: Fixed at eb06770e. `timing.mjs` now delegates Backbone and adapter import/configuration to `loadBackboneRuntime()`. The committed timing-harness digest and exact-head approval were refreshed for that source change.

File: docs/common.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ new MyObject({ foo: 'bar' }, 'baz'); // logs "bar" "bar" "baz"
 Marionette classes include Marionette's owned [Events API](./events.md). Each
 class can emit events and listen to other objects that implement the compatible
 event interface. Backbone interoperability is available through the explicit
-[`marionette/backbone` shim](./events.md#backbone-interop); the
+[`@marionette/adapters/backbone` integration](./events.md#backbone-interop); the
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply docs/common.md:50]: Fixed at eb06770e: the wording now says Backbone emitters participate after configuring the explicit integration.

File: test/performance/timing.test.mjs [hunk 7]
```diff
@@ -64,7 +64,7 @@ describe('hosted timing report math', () => {
     try {
       await assert.rejects(
         measure({ root: fixtureRoot, configPath, dependencyRoot: root }),
-        /dist\/backbone\.js/
+        /dist\/marionette\.js/
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/performance/timing.test.mjs]: Fixed at eb06770e. The rejection assertion now accepts either concurrently missing runtime path, while still proving DOM globals are restored for the failed load.

**Comment by paulfalgout** [comment]
**Approval stands** at `eb06770e`. The `disposeOwned` narrowing is right, and it is the kind of correction that is easy to talk yourself out of.

```diff
  disposeOwned(source) {
-   source.stopListening();
-   source.off();
+   void source;
  },
```

The previous version looked responsible — an owned source, released through Backbone's own cleanup methods rather than persistence-capable `destroy()`. But `off()` with no arguments removes **every** handler on that model, and `stopListening()` drops everything the model was observing. Neither is scoped to Marionette. A caller who did `owner.getState().on('change', …)` on a factory-created source would have that handler silently removed at owner teardown, by a framework that never registered it.

Doing nothing is the honest answer, and the doc change says so plainly rather than leaving the no-op to be discovered:

> The adapter leaves an owned Backbone state source and its caller-owned listeners intact because Backbone has no source-wide disposal operation that can preserve them.

Marionette still releases the subscriptions it created — it tracks those — so nothing leaks on Marionette's side. What is given up is disposal of the source object itself, which Backbone simply does not offer safely.

### One consequence worth recording where the contract lives

For the Backbone adapter, owned and borrowed state sources now behave **identically** at teardown. The #380 contract draws the distinction as *"A factory result is owned: Marionette also disposes it at owner destruction through the selected adapter contract"*, and the stable-release gate requires that *"factory results are disposed exactly once after subscription release."* For Backbone that gate is now vacuously true — there is nothing to dispose, so it cannot fail.

That is not wrong; it is the correct outcome of an adapter that cannot dispose its sources. But the ownership distinction is currently documented as a general contract in ROADMAP and the state docs, while its adapter-specific emptiness is documented only in `optional-backbone.md`. Someone reasoning from the contract will expect `createState()` to buy them disposal, and with the Backbone adapter it buys them nothing beyond subscription release.

A sentence where the ownership rule is stated — that disposal is adapter-dependent and some adapters cannot provide it — would keep the general contract honest without weakening it. This also matters for the release gate: it should be clear whether "disposed exactly once" means "disposed where the adapter supports disposal" or whether an adapter that cannot dispose fails the gate.

File: test/fixtures/backbone-adapter/validate.mjs [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { execFileSync } from 'node:child_process';
+import assert from 'node:assert/strict';
+import { existsSync } from 'node:fs';
+import { resolve } from 'node:path';
+
+assert.equal(existsSync(resolve(import.meta.dirname, 'node_modules/jquery')), false);
+await assert.rejects(
+  import('@marionette/adapters'),
+  error => error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED',
+);
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/fixtures/backbone-adapter/validate.mjs:13]: Fixed at adc24429. The packed fixture now uses `createRequire(import.meta.url)` and asserts that CommonJS `require("@marionette/adapters")` fails with `ERR_PACKAGE_PATH_NOT_EXPORTED`, alongside the existing ESM root-boundary assertion. The targeted exact-tarball fixture passes.

File: test/setup/vitest.js [hunk 9]
```diff
@@ -13,9 +13,9 @@ import Backbone from 'backbone';
 import sinon from 'sinon';
 import sinonChai from 'sinon-chai';
 import * as Marionette from '../../index.js';
-import BackboneDataApi from '../../runtime/backbone-data-api.js';
+import BackboneApi from '../../packages/adapters/src/backbone-api.js';
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/setup/vitest.js:16]: Keeping these as source imports intentionally. Vitest is the source-level suite and its 100% coverage contract includes `packages/adapters/src/**/*.js`; pointing it at built output would make source edits depend on a prior build and remove direct source coverage. Public ESM/CJS resolution is already exercised from exact packed tarballs, including the new Vite fixture and the three-browser packed-adapter test, so this change would duplicate boundary coverage while weakening the source suite.

File: test/unit/backbone-adapter.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/backbone-adapter.spec.js]: Fixed at eb06770e and retained at adc24429. `disposeOwned` is a no-op, the owner still runs its targeted subscription disposer, and the regression requires the external listener to fire after owner destruction. Package and guide text now state that caller-owned Backbone listeners remain intact.

**Comment by paulfalgout** [comment]
**Approval stands** at `adc24429`. Fixture-only; core aggregate remains **88,406** across its four artifacts.

Extending the no-root-barrel assertion to `require()` closes a real gap rather than adding symmetry for its own sake. `exports` resolves per condition, so a root entry can be absent under `import` and present under `require` — and this package ships both `.js` and `.cjs` builds, which is exactly the situation where the two can drift apart. Asserting only the ESM rejection left the CommonJS root unproven while the PR claims no root barrel at all.

`createRequire(import.meta.url)` inside an `.mjs` validator is the right way to test the other resolution mode from the same fixture, rather than adding a parallel `.cjs` file that could fall out of sync with this one.

This is the same lesson as the `.d.mts` → `.d.ts` fix in #384: a packaging guarantee holds per module system, and verifying one says nothing about the other. Good to see it applied without needing the bug first this time.
